### DISCUSSION TITLE
Add an external call to the Reranker model through the OpenAI API in the dashboard

### DISCUSSION THREAD

**Discussion by andyhau520** [open-webui/open-webui#7636] [Ideas]
Hope to add external calls to the Reranker model through the OpenAI API in the dashboard to reduce the resource consumption of openwebui itself.

**Comment by oatmealm** [comment]
Really looking forward to this. Also would be great to have the reranker exposed as an endpoint!

**Reply by reecelikesramen** [reply]
Also looking forward to this. I want to use Jina

**Comment by audrzejq** [comment]
Would be so great! Looking for it!

**Comment by mkhludnev** [comment]
Turns out OpenAI has no /rerank api, however here's how to use gpts for reranking 
 - https://cookbook.openai.com/examples/question_answering_using_a_search_api 
 - https://cookbook.openai.com/examples/search_reranking_with_cross-encoders

Conventional APIs are listed here:
 - https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html?ref=blog.mozilla.ai#supported-apis
 - + https://localai.io/features/reranker/

I want to check some reranker in langchain https://python.langchain.com/docs/integrations/document_transformers/

Let's also link https://github.com/open-webui/open-webui/discussions/3317
